lib/zoho_resource_client.rb

```diff
@@ -0,0 +1,73 @@
+class ZohoResourceClient
+  ZOHO_ORG_ID = ArchiveConfig.ZOHO_ORG_ID
+  CONTACT_SEARCH_ENDPOINT = 'https://desk.zoho.com/api/v1/contacts/search'.freeze
+  CONTACT_CREATE_ENDPOINT = 'https://desk.zoho.com/api/v1/contacts'.freeze
+  TICKET_CREATE_ENDPOINT = 'https://desk.zoho.com/api/v1/tickets'.freeze
+
+  def initialize(access_token:, email:, username: nil)
+    @access_token = access_token
+    @email = email
+    @username = username
+  end
+
+  def retrieve_contact_id
+    (find_contact || create_contact).fetch('id')
```

Comment:
The Zoho docs have a warning about searching for brand new records:
> New resources that you add to your help desk might take some time to be indexed for search. Therefore, if the resources do not appear in search results, wait for a few minutes and try again.

Have you tried submitting two reports at almost exactly the same time with the same (never-before-used) email address? Does it create duplicate contact records? Does it generate an error when trying to create the second contact? If it does generate an error, should that error be handled in some way? Or is the indexing too fast to run into that sort of issue at the moment?

Reply:
No idea what happens, thanks for catching that! Ideas welcome 😬